utils/compression/zstd_compressor.go

```diff
@@ -0,0 +1,24 @@
+// Copyright (C) 2019-2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package compression
+
+import (
+	"github.com/DataDog/zstd"
+)
+
+var _ Compressor = (*zstdCompressor)(nil)
+
+func NewZstdCompressor() Compressor {
+	return &zstdCompressor{}
+}
+
+type zstdCompressor struct{}
+
+func (*zstdCompressor) Compress(msg []byte) ([]byte, error) {
+	return zstd.Compress(nil, msg)
+}
+
+func (*zstdCompressor) Decompress(msg []byte) ([]byte, error) {
+	return zstd.Decompress(nil, msg)
+}
```

Comment:
This only works for message sizes up to 1MiB - which breaks the current max message size.

Reply:
I just added a test and this does seem to work for 2 MiB messages

Reply:
I don't fully understand how this function works so I suppose it's still possible for this to fail if `hint` is smaller than `upperBound`:

```go
func decompressSizeHint(src []byte) int {
	// 1 MB or 10x input size
	upperBound := 10 * len(src)
	if upperBound < decompressSizeBufferLimit {
		upperBound = decompressSizeBufferLimit
	}

	hint := upperBound
	if len(src) >= zstdFrameHeaderSizeMax {
		hint = int(C.ZSTD_getFrameContentSize(unsafe.Pointer(&src[0]), C.size_t(len(src))))
		if hint < 0 { // On error, just use upperBound
			hint = upperBound
		}
	}

	// Take the minimum of both
	if hint > upperBound {
		return upperBound
	}
	return hint
}
```

Reply:
Err actually I'm not so sure -- it looks like if `dst` in `zstd.Decompress` is too small then it'll fall back to using its stream API which I think (based on their docs) works regardless of the size of the decompressed message.